ckanext/cfpb_extrafields/validators.py

```diff
@@ -117,3 +117,15 @@ def clean_select_multi(s):
         clean.append(s[left:])
     return clean
 
+
+ROLE_PREFIX = "db_role_level_"
+DESC_PREFIX = "db_desc_level_"
+def combine_roles(data):
+    roles = sorted((key[len(ROLE_PREFIX):], val) for key, val in data.items() if key.startswith(ROLE_PREFIX) and val)
+    if roles:
+        data["db_roles"] = [(role, data.get(DESC_PREFIX + num, "")) for num, role in roles]
+
+    items_to_delete = [key for key in data.keys() if key.startswith(ROLE_PREFIX) or key.startswith(DESC_PREFIX)]
+    for key in items_to_delete:
+        del data[key]
+    return data
```

Comment:
This is a clever way of handling the change in data structure :)  I think ultimately we'll want to do away with the old fixed roles, but it does make sense to do that as a later step.